core/sile.lua

```diff
@@ -286,15 +291,50 @@ SILE.currentCommandStack = {
       str = str .. type(self.content) .. "(" .. self.content .. ")"
     end
 
-    --str = str .. "\n\n" .. self.pushTrace .. "\n"
-
     return str
+  end,
+  --- Internal: Given an collection of frames and an afterFrame, construct and return a human readable info string
+  --- about the location in processed document. Similar to _frameToLocationString, but takes into account
+  --- the afterFrame and the fact, that not all frames may carry a location information.
+  _formatTraceHead = function (traceStack, afterFrame)
+    local top = traceStack[#traceStack]
+
+    if not top then
+      -- Stack is empty, there is not much we can do
+      return afterFrame and "after " .. afterFrame:toLocationString() or nil
+    end
+
+    local info = top:toLocationString()
+
+    -- Not all stack traces have to carry location information.
+    -- If the top stack trace does not carry it, find a frame which does.
+    -- Then append it, because its information may be useful.
+    if not top.line then
+      for i = #traceStack - 1, 1, -1 do
+        if traceStack[i].line then
+          -- Found a frame which does carry some relevant information.
+          info = info .. " near " .. traceStack[i]:toLocationString(--[[skipFile=]] traceStack[i].file == top.file)
+          break
+        end
+      end
+    end
+
+    -- Print after, if it is in a relevant file
+    if afterFrame and (not base or afterFrame.file == base.file) then
```

Comment:
I just noticed `base` is never set anywhere in the scope of this function, what condition was this supposed to catch?

Reply:
That was supposed to be `local` with the frame carrying the location information which I removed in the last refactor. I blame Lua's lack of compiler, will fix ASAP.

Reply:
I can do it, don't worry about it. Just checking for feedback here...